### PR TITLE
Rock mod_* - part 2 - mod_disco..mod_mam

### PR DESCRIPTION
**Pull request by fenek**
This is a second part of rocking `mod_*` with Elvis.

Only one issue will become unresolved for now: `mod_mam_utils` being a god module. I don't think decomposing a module with 50+ functions into smaller ones with ~20 functions is in scope of this PR.

### CODE REVIEW

File: apps/ejabberd/src/mod_http_notification.erl [hunk 1]
```diff
@@ -7,7 +7,11 @@
 
 -module(mod_http_notification).
 -author("baibossynov.valery@gmail.com").
+
 -behaviour(gen_mod).
+-behaviour(mod_http_notification).
```
Comment (gadgetci) [inline apps/ejabberd/src/mod_http_notification.erl]: According to **Compiler**:
> Warning: behaviour mod_http_notification undefined

File: apps/ejabberd/src/mod_last.erl [hunk 2]
```diff
@@ -145,41 +143,36 @@ now_to_seconds({MegaSecs, Secs, _MicroSecs}) ->
...
+-spec process_sm_iq(ejabberd:jid(), ejabberd:jid(), ejabberd:iq()) -> ejabberd:iq().
+process_sm_iq(_From, _To, #iq{type = set, sub_el = SubEl} = IQ) ->
+    IQ#iq{type = error, sub_el = [SubEl, ?ERR_NOT_ALLOWED]};
+process_sm_iq(From, To, #iq{type = get, sub_el = SubEl} = IQ) ->
+    User = To#jid.luser,
+    Server = To#jid.lserver,
+    {Subscription, _Groups} =
+    ejabberd_hooks:run_fold(roster_get_jid_info, Server,
+                            {none, []}, [User, Server, From]),
+    case (Subscription == both) or (Subscription == from) or
```
Comment (michalwski) [inline apps/ejabberd/src/mod_last.erl]: To be honest, I'm not sure how should I read all these conditions to validate if this is correct or not. Would it be possible to make easier to understand by f.e introduction a function(s) or assigning results to some meaningful names?

**Comment by fenek** [comment]
@michalwski 
I have to make it work first... :D

File: apps/ejabberd/src/mod_last.erl (hunk 2 above)
Comment (fenek) [inline apps/ejabberd/src/mod_last.erl]: Sure, not a problem.

**Pull request merged**